apps/snitch_core/priv/repo/seed/orders.ex

```diff
@@ -0,0 +1,151 @@
+  }
+
+  @order %{
+    slug: nil,
+    state: nil,
+    billing_address_id: nil,
+    shipping_address_id: nil,
+    user_id: nil,
+    adjustment_total: Money.new(0, :USD),
+    promo_total: Money.new(0, :USD),
+    item_total: Money.new(0, :USD),
+    total: Money.new(0, :USD),
+    inserted_at: DateTime.utc(),
+    updated_at: DateTime.utc()
+  }
+
+  @variant_count 5
+  @max_line_item_count 3
+
+  def seed_orders! do
+    variants = Repo.all(Variant)
+    [address | _] = Repo.all(Address)
+    [user | _] = Repo.all(User)
+    line_item_stream = random_line_items(variants)
+
+    digest = %{
+      cart: [user_id: user.id],
+      address: [user_id: user.id, address_id: address.id],
+      payment: [user_id: user.id, address_id: address.id]
+    }
+
+    {orders, line_items} = make_orders(digest, line_item_stream)
+
+    {_, order_structs} =
+      Repo.insert_all(
+        Order,
+        orders,
+        on_conflict: :nothing,
+        conflict_target: [:slug],
+        returning: [:id, :slug]
+      )
+
+    filtered_line_items =
+      order_structs
+      |> Enum.map(fn %{id: id, slug: slug} ->
+        line_items
+        |> Enum.reduce(&Map.merge/2)
+        |> Map.fetch!(slug)
+        |> Stream.map(&Map.put(&1, :order_id, id))
+        |> Enum.map(&Map.delete(&1, :slug))
+      end)
+      |> List.flatten()
+
+    Repo.insert_all(LineItem, filtered_line_items)
+  end
+
+  def make_orders(digest, line_item_stream) do
+    digest
+    |> Enum.map(fn {state, opts} ->
+      slug = "order-in-#{state}"
+
+      line_items =
+        line_item_stream
+        |> Enum.take(@max_line_item_count)
+        |> Enum.map(&Map.put(&1, :slug, slug))
+
+      item_total =
+        line_items
+        |> Stream.map(&Map.fetch!(&1, :total))
+        |> Enum.reduce(&Money.add!/2)
+
+      order = %{
+        @order
+        | slug: slug,
+          state: "#{state}",
+          user_id: opts[:user_id],
+          billing_address_id: opts[:address_id],
+          shipping_address_id: opts[:address_id],
+          item_total: item_total,
+          total: item_total
+      }
+
+      {order, %{slug => line_items}}
+    end)
+    |> Enum.unzip()
+  end
+
+  def random_line_items(variants) do
+    random_pick = Stream.repeatedly(fn -> Enum.random(variants) end)
+
+    Stream.map(random_pick, fn v ->
+      quantity = :rand.uniform(3)
+
+      %{
+        @line_item
+        | variant_id: v.id,
+          quantity: quantity,
+          unit_price: v.cost_price,
+          total: Money.mult!(v.cost_price, quantity)
+      }
+    end)
+  end
+
+  def seed_variants!(count \\ nil) do
+    Repo.insert_all(
+      Variant,
+      Enum.take(variants(), count || @variant_count),
+      returning: [:id],
+      on_conflict: :nothing
+    )
+  end
+
+  def variants do
+    0
+    |> Stream.iterate(&(&1 + 1))
+    |> Stream.map(&"shoes-nike-#{&1}")
+    |> Stream.map(fn sku -> %{random_variant() | sku: sku} end)
+  end
+
+  def random_variant do
+    %{
```

Comment:
Why are we not using `%Variant{}`

Reply:
Because [`Ecto.Repo.insert_all`](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_all/3) requires a `[map] | keyword`.